python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -648,6 +644,16 @@ def foo(param: Pattern) -> Pattern:
     assertThat(((FunctionType) functionDef.name().typeV2()).returnType().unwrappedType()).isEqualTo(patternType);
   }
 
+  @Test
+  void inferFunctionParameterTypes5() {
+    FileInput root = inferTypes("""
+      my_alias = int
+      def foo(param: my_alias): ...
+      """);
+
+    var functionDef = (FunctionDef) root.statements().statements().get(1);
+    assertThat(((FunctionType) functionDef.name().typeV2()).parameters().get(0).declaredType().unwrappedType()).isEqualTo(INT_TYPE);
```

Comment:
I'm surprised this works, that's pretty nice!
Now the question would be, what if it's assigned multiple times?
```
my_alias = int
my_alias = str
def foo(param: my_alias): ...
```
or even:
```
if cond:
    my_alias = int
else:
    my_alias = str
def foo(param: my_alias): ...
```